src/terrama2/services/analysis/core/BaseContext.cpp

```diff
@@ -127,12 +126,12 @@ terrama2::services::analysis::core::BaseContext::getRasterList(const terrama2::c
 
         auto cachedRaster = addRaster(key, dsRaster);
         auto interpolationMethod = static_cast<int>(analysis_->outputGridPtr->interpolationMethod);
-        std::shared_ptr<terrama2::core::SynchronizedInterpolator> syncInterpolator = std::make_shared<terrama2::core::SynchronizedInterpolator>(dsRaster.get(), interpolationMethod);
+        std::shared_ptr<terrama2::core::SynchronizedInterpolator> syncInterpolator = std::make_shared<terrama2::core::SynchronizedInterpolator>(cachedRaster.get(), interpolationMethod);
         interpolatorMap_.emplace(cachedRaster, syncInterpolator);
       }
     });
 
-    return rasterMap_[key];
+    return rasterMap_.at(key);
```

Comment:
There is no guarantee that it will be data available, so using at will raise an exception .